.travis.yml

```diff
@@ -1,56 +1,40 @@
-        # https://help.github.com/articles/creating-an-oauth-token-for-command-line-use
-        #
-        # curl -u 'username' -d '{"scopes":["repo"],"note":"push to gh-pages from travis"}' https://api.github.com/authorizations
-        #
-        # It must be encrypted using the travis gem
-        # http://about.travis-ci.org/docs/user/build-configuration/#Secure-environment-variables
-        #
-        # travis encrypt GH_OAUTH_TOKEN=XXXXXXXXXXXXXXX
-        #
-        # User specific env variables
-        - secure: "nH1/eONF+2NYv4ReM55XWsrsWOTjJjaJy+EIy0R4QiVqwMpT6WGwzihb/7fqWH3tnmnqsQECv76Fe0hHghsvRCw600WMTR8rAmsjwCJVsa3T9v+puHN/dXkRmuDaB0aoD8aE4kWAj2HjIMt5bQtzN8qs3W2y49/NkLLnebMgo8E="
-        - GH_OWNER: CruGlobal
-        - GH_PROJECT_NAME: conf-registration-web
-        # This .travis.yml file instructs travis-ci.org to build your master branch using `yeoman build`
-        # and deploy the output to your project's gh-pages branch.
-
-before_script:
-    # install dependencies
-    - gem update --system
-    - npm install -g grunt-cli bower
-    - if [ "${TRAVIS_PULL_REQUEST}" != "false" ]; then export GH_PUBLISH_TO_MODULE='build-stage'; export GH_COMMIT_MSG='deployable PR '$TRAVIS_PULL_REQUEST' to '$GH_PUBLISH_TO_MODULE; fi
-    - if [ "${TRAVIS_PULL_REQUEST}" == "false" ]; then export GH_PUBLISH_TO_MODULE='gh-pages'; export GH_COMMIT_MSG='deployable master to '$GH_PUBLISH_TO_MODULE; fi
+sudo: false
+language: node_js
+node_js: '7'
 script:
-    # We want to gate on passing tests and a successful build
-    - bower install
-    - if [ "${TRAVIS_PULL_REQUEST}" != "false" ]; then grunt build:stage; fi
-    - if [ "${TRAVIS_PULL_REQUEST}" == "false" ]; then grunt build:dist; fi
+- yarn run test -- --single-run
+- if [ "$TRAVIS_PULL_REQUEST" = "false" ] && ([ "$TRAVIS_BRANCH" == "master" ]
+  || [ "$TRAVIS_BRANCH" == "staging" ]); then yarn run build; fi
 after_success:
-    - git submodule add -b ${GH_PUBLISH_TO_MODULE} https://${GH_OAUTH_TOKEN}@github.com/${GH_OWNER}/${GH_PROJECT_NAME} site > /dev/null 2>&1
-    - cd site
-    - if git checkout ${GH_PUBLISH_TO_MODULE}; then git checkout -b ${GH_PUBLISH_TO_MODULE}; fi
-    - git rm -r .
-    - cp -R ../dist/* .
-    - cp ../dist/.* .
-    - git add -f .
-    - git config user.name "Travis CI"
-    - git config user.email "$COMMIT_AUTHOR_EMAIL"
-    - git commit -am "${GH_COMMIT_MSG}"
-    - git push https://${GH_OAUTH_TOKEN}@github.com/${GH_OWNER}/${GH_PROJECT_NAME} HEAD:${GH_PUBLISH_TO_MODULE} > /dev/null 2>&1
-before_install:
-    - "export DISPLAY=:99.0"
-    - "sh -e /etc/init.d/xvfb start"
+- bash <(curl -s https://codecov.io/bash)
+cache:
+  yarn: true
+  directories:
+  - node_modules
+before_deploy:
+  - pip install --user awscli
+deploy:
+- provider: script
+  script: ~/.local/bin/aws s3 sync dist s3://cru-ert-web-stage --region us-east-1 --acl public-read --exclude '*.html' --cache-control 'public, max-age=31536000'
```

Comment:
If the file isn't in cloudfront, it will hit s3 directly.  o yes, the separate requests would be a problem.  It would probably delete files from the previous request.  I guess it's fine.  Just might have many files over time.